CMakeLists.txt

```diff
@@ -46,6 +46,13 @@ if (WIN32)
     add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
 endif()
 
+if (NOT GGML_CUDA)
```

Comment:
What is the reason for disabling this with `GGML_CUDA`? Shouldn't these changes only apply to C/C++ sources anyway?

Reply:
It's because these changes will cause cuda failed to build.

Reply:
Does this work?
```
if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
    add_compile_options("$<$<COMPILE_LANGUAGE:C>:/source-charset:utf-8>")
    add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/source-charset:utf-8>")
    add_compile_options("$<$<COMPILE_LANGUAGE:C>:/execution-charset:utf-8>")
    add_compile_options("$<$<COMPILE_LANGUAGE:CX>:/execution-charset:utf-8>")
endif()
```

Reply:
It works.

Reply:
It works.